src/main/java/com/chopshop166/chopshoplib/sensors/CSFusedEncoder.java

```diff
@@ -38,13 +40,18 @@ public void reset() {
         System.out.println("Reseting Fused Encoder");
         relativeEncoderOffset = this.absolutePos.getAbsolutePosition();
         this.relativeEncoder.reset();
+        this.motorEncoder.reset();
     }
 
     @Override
     public double getDistance() {
 
         double distance = this.relativeEncoder.getDistance() + relativeEncoderOffset;
+        double motorDistance = this.motorEncoder.getDistance();
         // Keep trying to set the offset until we get a valid reading back
+        if (distanceExceedsThreshold(distance, motorDistance)) {
```

Comment:
We're probably going to have to be smarter than this here unfortunately. The motor encoder readings seem to be much higher at higher angles for some reason. They also can go quite a bit negative where the absolute encoder stops changing. We probably want this check to look at the rates and if the motor encoder is moving but the relative encoder isn't them switch. With a special exception for when the angle is below 1 degree(ish)

It should also store the current state (trusting the relative encoder or not) and provide a way to get that value, and set it.

Reply:
Ok I'm going to push the first two, then figure out this one